packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -46,11 +50,19 @@ class ImageInfo {
   /// A string used for debugging purpopses to identify the source of this image.
   final String? debugLabel;
 
+  /// Whether or not the [image] should be disposed of when an associated
+  /// [ImageStream] loses its last listener.
+  ///
+  /// If multiple [ImageStream]s may refer to the same image, this should be set
+  /// to true. Within the framework, this does not normally occur, but custom
+  /// [ImageProvider] or [ImageStream] implementations can behave differently.
+  final bool keepAlive;
```

Comment:
does re-using the framework's "keepAlive" nomenclature here have some potential for confusion? i.e. if users think that keep alive in the framework might be necessary for perf/state, might naming this similarly lead them to believe this should also be true?

Reply:
In my mind it's a similar trade-off.

The only users who should be dealing with this are ones who are writing their own image providers, streams, or caches. If you think it'd help to expand the documentation here I'd be happy to, or if there's a clearer name for this that would be less confusing I'm not stuck on it.

Reply:
you could sort of invert the naming here - use it to describe what additional action to be taken. Instead of keeping it alive, you are eagerly disposing it?